inference-engine/src/transformations/include/transformations/low_precision/layer_transformation.hpp

```diff
@@ -0,0 +1,380 @@
+        return signedInterval ? element::i8 : element::u8;
+    }
+
+    static float getMin(const size_t quantizationLevels, const bool signedInterval) {
+        if (quantizationLevels == 255) {
+            return signedInterval  ? -127.0 : 0.0;
+        } else if (quantizationLevels == 256) {
+            return signedInterval ? -128.0 : 0.0;
+        } else {
+            // THROW_TRANSFORMATION_EXCEPTION << "quantization level " << quantizationLevels << " is not supported";
+            // FIXME: not completed
+            return signedInterval ? -128.0 : 0.0;
+        }
+    }
+
+    static float getMax(const size_t quantizationLevels, const bool signedInterval) {
+        if ((quantizationLevels == 255) || (quantizationLevels == 256)) {
+            return signedInterval ? 127.0 : 255.0;
+        } else {
+            // THROW_TRANSFORMATION_EXCEPTION << "quantization level " << quantizationLevels << " is not supported";
+            // FIXME: not completed
+            // return quantizationLevels - 1.0;
+            return signedInterval ? 127.0 : 255.0;
+        }
+    }
+};
+
+inline bool operator==(const DataPrecision& value1, const DataPrecision& value2) {
+    return
+            (value1.precision == value2.precision) &&
+            (value1.min == value1.min) &&
+            (value1.max == value1.max);
+}
+
+inline bool operator!=(const DataPrecision& value1, const DataPrecision& value2) {
+    return !(value1 == value2);
+}
+
+inline std::ostream &operator << (std::ostream &os, const DataPrecision& value) {
+    os << value.precision << ", min: " << value.min << ", max: " << value.max;
+    return os;
+}
+
+// Base class for all LP transformations, holds some common data structures
+class TRANSFORMATIONS_API LayerTransformation {
+public:
+    enum QuantizedTensorAlignment {
+        None,
+        UpdateLevel
+    };
+
+    class Params {
+    public:
+        Params(
+                const bool updatePrecisions = true,
+                const QuantizedTensorAlignment quantizedTensorAlignmentOnActivations = QuantizedTensorAlignment::UpdateLevel,
+                const QuantizedTensorAlignment quantizedTensorAlignmentOnWeights = QuantizedTensorAlignment::None,
+                bool supportAsymmetricQuantization = false,
+                std::vector<element::Type> precisionsOnActivations = { element::u8, element::i8 },
+                std::vector<element::Type> precisionsOnWeights = { element::i8 }) :
+                updatePrecisions(updatePrecisions),
+                quantizedTensorAlignmentOnActivations(quantizedTensorAlignmentOnActivations),
+                quantizedTensorAlignmentOnWeights(quantizedTensorAlignmentOnWeights),
+                supportAsymmetricQuantization(supportAsymmetricQuantization),
+                precisionsOnActivations(precisionsOnActivations),
+                precisionsOnWeights(precisionsOnWeights) {
```

Comment:
I think to split into default constructor and constructor with parameters will be good idea. The constructor signature with all optional arguments is confusing a little bit.